controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -199,6 +216,27 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 				break
 			}
 		}
+
+		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		if r.cephBlockPool.Name == "" {
+			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+			listOptions := &client.MatchingLabels{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
+			}
+			if err := r.list(cephBlockPoolList, client.InNamespace(r.OperatorNamespace), listOptions); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// if we find only one CephBlockPool with our query, we're good
+			// if we find more than one, we can't determine which one to select, so error out
+			if len(cephBlockPoolList.Items) == 1 {
+				r.cephBlockPool.Name = cephBlockPoolList.Items[0].GetName()
+			} else if len(cephBlockPoolList.Items) > 1 {
+				return reconcile.Result{}, fmt.Errorf("too many CephBlockPools found for storage consumer '%s' and storage profile '%s'", r.storageConsumer.Name, profileName)
```

Comment:
```suggestion
return reconcile.Result{}, fmt.Errorf("too many CephBlockPools found for storage consumer %q and storage profile %q", r.storageConsumer.Name, profileName)
```

Reply:
"Too many" is generic, can we just specify the number of CBP found?
```suggestion
				return reconcile.Result{}, fmt.Errorf(
					"%d CephBlockPools found matching the storage profile '%s'  requested for consumer  '%s'", 
					len(cephBlockPoolList.Items),
					profileName, 
					r.storageConsumer.Name
				)
```